### PR TITLE
reflect: remove manual `Reflect` impls which could be handled by macros

### PR DESCRIPTION
**Pull request by soqb**
# Objective

- In some cases we used manual impls for certain types, though they are (at least, now) unnecessary.

## Solution

- Use macros and reflecting-by-value to avoid this clutter.
  - Though there were linker issues with `Reflect` and the `CowArc` in `AssetPath` (see #9747), I checked these are resolved by using `#[reflect_value]`.

### CODE REVIEW

**Review by alice-i-cecile — APPROVED** [review]

File: crates/bevy_asset/src/path.rs [hunk 1]
```diff
@@ -52,7 +47,8 @@ use thiserror::Error;
 /// This means that the common case of `asset_server.load("my_scene.scn")` when it creates and
 /// clones internal owned [`AssetPaths`](AssetPath).
 /// This also means that you should use [`AssetPath::parse`] in cases where `&str` is the explicit type.
-#[derive(Eq, PartialEq, Hash, Clone, Default)]
+#[derive(Eq, PartialEq, Hash, Clone, Default, Reflect)]
+#[reflect_value(Debug, PartialEq, Hash, Serialize, Deserialize)]
```
Comment (MrGVSV) [inline crates/bevy_asset/src/path.rs:51]: We don't have to update it in this PR, but any reason why this is a `reflect_value` type? I would have assumed `CowArc` would be reflectable, but maybe not?

File: crates/bevy_render/src/render_asset.rs [hunk 2]
```diff
@@ -67,7 +63,8 @@ bitflags::bitflags! {
     /// If you have an asset that doesn't actually need to end up in the render world, like an Image
     /// that will be decoded into another Image asset, use `MAIN_WORLD` only.
     #[repr(transparent)]
-    #[derive(Serialize, TypePath, Deserialize, Hash, Clone, Copy, PartialEq, Eq, Debug)]
+    #[derive(Serialize, Deserialize, Hash, Clone, Copy, PartialEq, Eq, Debug, Reflect)]
```
Comment (MrGVSV) [inline crates/bevy_render/src/render_asset.rs:66]: We should probably register these traits

**Review by MrGVSV — COMMENTED** [review]
Amazing! Just one comment (and one non-blocking question)

**Review by james7132 — APPROVED** [review]
LGTM sans @MrGVSV's comments.

File: crates/bevy_render/src/render_asset.rs (hunk 2 above)
Comment (james7132) [reply crates/bevy_render/src/render_asset.rs:66]:
```suggestion
    #[derive(Serialize, Deserialize, Hash, Clone, Copy, PartialEq, Eq, Debug, Reflect)]
    #[reflect(Serialize, Deserialize, Hash, PartialEq, Debug, Reflect)]
```

File: crates/bevy_render/src/render_asset.rs [hunk 3]
```diff
@@ -67,7 +63,9 @@ bitflags::bitflags! {
...
+    #[reflect(Serialize, Deserialize, Hash, PartialEq, Debug, Reflect)]
```
Comment (MrGVSV) [inline crates/bevy_render/src/render_asset.rs:67]: How is CI passing if we never import `ReflectSerialize` et al? 🤔

**Comment by alice-i-cecile** [comment]
@soqb can you resolve merge conflicts? They're not trivial so I'm not going to tackle them in the web client.

**Comment by soqb** [comment]
not this week i'm afraid, but i do plan on it when there's time.

**Comment by OneFourth** [comment]
I'm interested in adopting this PR, I think I can fix the merge conflicts at least

**Pull request closed without merging**